### PR TITLE
ci: init changeset + github action

### PR DESCRIPTION
**Pull request by ayoubqrt**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
```
Comment (Yanzi-dev) [inline .github/workflows/publish.yml]: Please consider use `actions/checkout@v4`

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+      - name: Setup Node.js 18.x
+        uses: actions/setup-node@v3
```
Comment (Yanzi-dev) [inline .github/workflows/publish.yml]: Please consider `actions/setup-node@v4`

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        with:
+          node-version: 18.x
```
Comment (Yanzi-dev) [inline .github/workflows/publish.yml]: Please consider `          node-version: 20`

**Review by Yanzi-dev — CHANGES_REQUESTED** [review]
Great job @ayoubqrt please consider those suggestions

**Comment by ayoubqrt** [comment]
thanks, I modified the gh action according to your review

**Pull request merged**